packages/trace-viewer/src/ui/uiModeView.tsx

```diff
@@ -336,9 +323,26 @@ export const UIModeView: React.FC<{}> = ({
 
   // Watch implementation.
   React.useEffect(() => {
-    if (!testServerConnection)
+    if (!testServerConnection || !teleSuiteUpdater)
       return;
-    const disposable = testServerConnection.onTestFilesChanged(params => {
+    const disposable = testServerConnection.onTestFilesChanged(async params => {
```

Comment:
Let's comment that we first update the list to get the up-to-date list of tests, and then run the watched ones.